app/services/order_service.rb

```diff
@@ -0,0 +1,53 @@
+class OrderService
+  def initialize(order = nil)
+    @order = order
+  end
+
+  def self.checkout(user, payment_method:)
+    new.checkout(user, payment_method: payment_method)
+  end
+
+  def checkout(user, payment_method:)
+    cart_items = user.cart_items.includes(:food_item)
+    return { success: false, message: "Cart is empty" } if cart_items.blank?
+
+    order = nil
+    begin
+      Order.transaction do
+        order = user.orders.create!(
+          status: :preparing,
+          payment_method: payment_method,
+          total_price: 0
+        )
+        total = 0
+        cart_items.each do |ci|
+          unit_price = ci.food_item.price
+          order.order_items.create!(
+            food_item: ci.food_item,
+            quantity: ci.quantity,
+            unit_price: unit_price
+          )
+          total += unit_price * ci.quantity
+        end
+
+        order.update!(total_price: total)
```

Comment:
You have the same logic in the other service. Perhaps create another one to handle this or a helper.